tableau_blueprints/job_status.py

```diff
@@ -0,0 +1,132 @@
+EXIT_CODE_STATUS_INCOMPLETE = 210
+EXIT_CODE_JOB_NOT_FOUND = 404
+EXIT_CODE_JOB_CACNELLED = 403
+EXIT_CODE_GENERIC_QUERY_JOB_ERROR = 400
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--username', dest='username', required=True)
+    parser.add_argument('--password', dest='password', required=True)
+    parser.add_argument('--site-id', dest='site_id', required=True)
+    parser.add_argument('--server-url', dest='server_url', required=True)
+    parser.add_argument('--job-id', dest='job_id', required=True)
+    args = parser.parse_args()
+    return args
+
+
+def authenticate_tableau(username, password, site_id, server_url):
+    """TSC library to sign in and sign out of Tableau Server and Tableau Online.
+
+    :param username:The name of the user.
+    :param password:The password of the user.
+    :param site_id: The site_id for required datasources. ex: ffc7f88a-85a7-48d5-ac03-09ef0a677280
+    :param server_url: This corresponds to the contentUrl attribute in the Tableau REST API.
+    :return: connection object
+    """
+    try:
+        tableau_auth = TSC.TableauAuth(username, password, site_id=site_id)
+        server = TSC.Server(server_url, use_server_version=True)
+        connection = server.auth.sign_in(tableau_auth)
+    except Exception as e:
+        print(f'Failed to connect to Tableau.')
+        print(e)
+        sys.exit(EXIT_CODE_INVALID_CREDENTIALS)
+    return server, connection
+
+
+def determine_run_status(run_details_response):
+    """Job status response handler.
+    The finishCode indicates the status of the job: 0 for success, 1 for error, or 2 for cancelled.
+
+    :param run_details_response:
+    :return:
+    """
+    run_id = run_details_response.id
+    if run_details_response.finish_code == 0:
+        if run_details_response.progress == "Pending":
+            print(f'Tableau reports that the run {run_id} in Pending status.')
+            exit_code = EXIT_CODE_STATUS_INCOMPLETE
+        elif run_details_response.progress == "Cancelled":
+            print(f'Tableau reports that run {run_id} was cancelled.')
+            exit_code = EXIT_CODE_JOB_CACNELLED
+        elif run_details_response.progress == "Failed":
+            print(f'Tableau reports that run {run_id} was Failed.')
+            exit_code = EXIT_CODE_FINAL_STATUS_ERRORED
+        elif run_details_response.progress == "InProgress":
+            print(f'Tableau reports that run {run_id} is in InProgress.')
+            exit_code = EXIT_CODE_STATUS_INCOMPLETE
+        else:
+            print(f'Tableau reports that run {run_id} was successful.')
+            exit_code = EXIT_CODE_FINAL_STATUS_SUCCESS
+    elif run_details_response.finish_code == 1:
+        print(f'Tableau reports that the job {run_id} is exited with error.')
+        exit_code = EXIT_CODE_GENERIC_QUERY_JOB_ERROR
+    else:
+        print(f'Tableau reports that the job {run_id} was cancelled.')
+        exit_code = EXIT_CODE_STATUS_INCOMPLETE
```

Comment:
What's the different between lines 64 and 79. Why are both using the same print statement but different resulting exit codes?